### ISSUE TITLE
Please add Retrode 2 autoconfig file

### ISSUE THREAD

**Issue by revvv** [libretro/retroarch-joypad-autoconfig#526]
The [Retrode 2](https://www.retrode.com) is an USB adapter for two original SNES gamepads.
My config file `Retrode.cfg` works fine on Windows: `autoconfig/xinput/Retrode.cfg`

```
input_driver = "hid"
input_device = "Retrode"
input_vendor_id = "1027"
input_product_id = "38849"
input_b_btn = "0"
input_y_btn = "1"
input_select_btn = "2"
input_start_btn = "3"
input_up_axis = "-1"
input_down_axis = "+1"
input_left_axis = "-0"
input_right_axis = "+0"
input_a_btn = "4"
input_x_btn = "5"
input_l_btn = "6"
input_r_btn = "7"
```
NOTE: I used `hid `as `input_driver ` which obviously doesn't play a role. My hope was that is will work on the Wii version, too. Which doesn't. Right now Wii USB Gamepad support is broken anyway: [(Wii) HID driver not working since 1.7.1](https://github.com/libretro/RetroArch/issues/7015)

**Comment by hizzlekizzle** [comment]
So, does it go into 'hid' or 'xinput'? Can you add the button labels, too? (I mean, they should be the same as the retropad assignments, but just to be complete)

**Comment by revvv** [comment]
Thanks for your quick reply! Please put it into `xinput`. I also changed the `input_driver` to avoid confusion. Button labels added as well:

```
input_driver = "xinput"
input_device = "Retrode"
input_vendor_id = "1027"
input_product_id = "38849"
input_b_btn = "0"
input_y_btn = "1"
input_select_btn = "2"
input_start_btn = "3"
input_up_axis = "-1"
input_down_axis = "+1"
input_left_axis = "-0"
input_right_axis = "+0"
input_a_btn = "4"
input_x_btn = "5"
input_l_btn = "6"
input_r_btn = "7"

input_b_btn_label = "B"
input_y_btn_label = "Y"
input_select_btn_label = "Select"
input_start_btn_label = "Start"
input_a_btn_label = "A"
input_x_btn_label = "X"
input_l_btn_label = "L"
input_r_btn_label = "R"

input_up_axis_label = "Dpad Up"
input_down_axis_label = "Dpad Down"
input_left_axis_label = "Dpad Left"
input_right_axis_label = "Dpad Right"
```

As I don't know the standard procedure, I'm not sure if you need to duplicate the file under `hid ` to support Linux etc. which I have not tested.

**Comment by hizzlekizzle** [comment]
Hmm, looking at a picture of a retrode, I'm concerned about the SNES vs Genesis/MD mapping. Do you have a 6-button Genesis/MD pad to compare with? Does this mapping for okay for that, too?

**Comment by revvv** [comment]
Unfortunately I don't have a Genesis controller, but a Competition Pro 5000: the D-PAD and Fire have the same mapping. (Fire = B)

I think this is why I omitted the labels in my first post.

Without the autoconfig file, you just see "Retrode not configured" on Windows. Of course you can configure it by yourself, but my hope was that one day it'll work on Wii version, too. Currently it doesn't even allow you to configure the buttons yourself. Simply not available.

**Comment by hizzlekizzle** [comment]
Ok, no worries. We'll go with this and if problems arise, we'll handle them as they come :)

**Comment by revvv** [comment]
Works perfectly! Thank you very much 🥇

**Comment by revvv** [comment]
Sorry, but just one more thing: I don't know if it's a good idea to copy the configuration file to hid directory, too, so that other platforms could benefit?

On Windows it doesn't work, but maybe I have used wrong combination (driver=raw, gamepad=hid)

**Comment by revvv** [comment]
I couldn't sleep, so I booted an Ubuntu Live Linux and created the autoconfiguration files.
Looks like Linux knows which gamepad port is used, so that we _could_ label SNES and Mega Drive  buttons correctly. " Right now I have no Mega Drive gamepad, so I just provided the same labels as for an SNES.

The attached zip contains all configuration files for Linux. The only difference is the `input_device`:
```
"Retrode Mega Drive #1"
"Retrode Mega Drive #2" 
"Retrode SNES / N64 #1"
"Retrode SNES / N64 #2"
```

[udev.zip](https://github.com/libretro/retroarch-joypad-autoconfig/files/3043138/udev.zip)
